sign/dilithium/gen.go

```diff
@@ -139,6 +143,9 @@ var (
 			DoubleEtaBits:  3,
 			Beta:           175,
 			Omega:          120,
+		}, {
+			Name:   "EdDilithium3",
```

Comment:
Probably to  preempt your question, @armfazh, I guess you'd prefer Edwards25519-Dilithium3 here?

Reply:
why is needed a new implementation of Mode3?

Reply:
It's not a full new implementation.  This extra line only tells `gen.go` to generate the file that registers a new mode.

Reply:
The hybrid mode is external to the scope of this package. There is no need to mantain a register of external implementations.

Reply:
I've considered putting the eddilithium3 as a subpackage under Dilithium.  The point is that in deployment you should definitely not  use Dilithium — you want to combine it with some classical algorithm.  Thus it makes sense from a usability perspective to include that immediately so that people don't have to write it themselves.

Reply:
Say some other deployment (not us) wants to use Dilithium+RSA, then following your argument, does every new deployent should modify the internals of Dilithium package? 
It isn't better to create a standalone package that fetches `crypto/rsa` and `circl/sign/dilithium` as dependencies.

Reply:
My proposal to include the Ed25519-Dilithium3 mode inside the Dilithium package does not prevent a third party from adding RSA2048-Dilithium3 using their own package.  They won't have the convenience of using the Modes API in the Dilithium package, but we won't either if we don't include the Ed25519-Dilithium3 mode in there.  So I do not see a downside.

Reply:
> They won't have the convenience of using the Modes API in the Dilithium package,

I didn't get this sentence. So, are you saying that having the hybrid mode inside the package has better usability than being in an external package? If so, that feels like the dilithium must improves its usability.

Reply:
In the meantime, could you please refactor to have the hybrid mode as an external package.

Reply:
> > They won't have the convenience of using the Modes API in the Dilithium package,
> 
> I didn't get this sentence. So, are you saying that having the hybrid mode inside the package has better usability than being in an external package? If so, that feels like the dilithium must improves its usability.

I think it should be easy to switch between Ed25519-Dilithium3 and Dilithium3, do you agree?

Reply:
> In the meantime, could you please refactor to have the hybrid mode as an external package.

It is already an external package.  What you're saying is that I should remove the ability to pick Ed25519-Dilithium3 via the dilithium.ModyByName API?

Reply:
After discussion I've removed EdDilithium3 support for the Dilitihium mode API.  The easy switching benefit will return when we have a unified signature/verifier API for all schemes.